File: 1548-check-if-all-1s-are-at-least-length-k-places-away/1548-check-if-all-1s-are-at-least-length-k-places-away.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        q = collections.deque()
        n = len(nums)

        for i in range(n):
            if nums[i] == 0:
                continue
            
            if q and i - q.popleft() <= k:
                return False
            
            if nums[i] == 1:
                q.append(i)
        
        return True

File: 1548-check-if-all-1s-are-at-least-length-k-places-away/1548-check-if-all-1s-are-at-least-length-k-places-away.cpp
class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        int n = nums.size();
        vector<int> vec;
        for (int i = 0; i < n; ++i)
            if (nums[i])
                vec.push_back(i);
        for (int i = 0; i + 1 < size(vec); ++i) {
            if (vec[i + 1] - vec[i] <= k)
                return false;
        }
        return true;
    }
};